File: introducao-a-computacao/2022/trabalho-pratico/csharp/domain/ResultadoContagem.cs
namespace IntroComp.Domain
{
    public readonly struct ResultadoContagem
    {
        public int Pares { get; }
        public int Impares { get; }

        public ResultadoContagem(int pares, int impares)
        {
            Pares = pares;
            Impares = impares;
        }
    }
}

File: introducao-a-computacao/2022/trabalho-pratico/java/src/introcomp/domain/ResultadoContagem.java
package introcomp.domain;

import java.util.Objects;


public final class ResultadoContagem {

    private final int pares;
    private final int impares;

    public ResultadoContagem(int pares, int impares) {
        if (pares < 0 || impares < 0) {
            throw new IllegalArgumentException("pares e impares devem ser não negativos");
        }
        this.pares = pares;
        this.impares = impares;
    }

    public int pares() {
        return pares;
    }

    public int impares() {
        return impares;
    }
}
